### PR TITLE
#161152337 fix header error and add 404 page

### PR DESCRIPTION
**Pull request by gonuoha**
#### What does this PR do?
fix bugs with frontend and add 404 page

#### Description of Task to be completed?
- Fix Header to Display Different Menu for Logged in user and Non-logged in user
- Horizontally Align the Search, Login and Register button properly
- Add Dropdown to the Header, so when a user clicks the avatar it displays
- Implement a 404 page for unknown routes

#### How should this be manually tested?
Run 'npm install'
Run 'npm run start'
view interface on the browser

#### What are the relevant pivotal tracker stories?
[#161152337](https://www.pivotaltracker.com/story/show/161152337)

### CODE REVIEW

**Review by emekafredy — CHANGES_REQUESTED** [review]
Hi @ozezzy. 
After checking out your work, I believe you did a tremendous job. The 404 page looks good and the redirection is appropriate. 
On the menu bar for a logged in user;
The cursor should show a 'pointer' ({cursor: pointer}) when hovering on the avatar, so the users know they can click on it to get some options. The same style should be applied to the login and signup buttons.
The publish option should only be available for users with the role of 'author'. Currently, users can see the publish sign on their menu bar and in the drop-down options.
On line 45 of the Menu.js component, you used 'lassName' instead of 'className'

File: src/components/Header/Menu.js [hunk 1]
```diff
@@ -1,28 +1,73 @@
...
+    const { triggerCategory, mobileCategory } = this.props;
+    triggerCategory(mobileCategory === 'visible' ? 'hidden' : 'visible');
+  }
+
+  render() {
+
+    const { user, mobileCategory } = this.props;
+    const { profileMenuVisibility } = this.state;
+    const { username, role } = user;
+    console.log(role)
```
Comment (lauragift21) [inline src/components/Header/Menu.js]: You might want to remove this concole.log statement on line 34.

**Review by lauragift21 — CHANGES_REQUESTED** [review]

File: src/AppRoutes.js [hunk 2]
```diff
@@ -10,18 +10,26 @@ import ProtectedRoute from './containers/ProtectedRoute';
...
+      <Switch>
+        <Route exact path='/' component={Homepage} />
+        {/* protected routes can be rendered using the ProtectedRoutes component */}
+        <Route path='/categories' component={AllCategories} />
+        <Route path='/category/:name' component={ArticleCategory} />
+        <ProtectedRoute path='/articles/:slug' component={ArticlePage} />
+        <ProtectedRoute path='/profile/user' component={ProfilePage} />
+        <Route exact path='/signin' component={Signin} />
+
+        {/* will alsways render when no other path is matched */}
```
Comment (emmaadesile) [inline src/AppRoutes.js]: alsways should be always

**Review by lauragift21 — APPROVED** [review]

File: src/components/Header/header.module.scss [hunk 3]
```diff
@@ -7,10 +7,13 @@ header {
   padding: 0 80px;
   box-sizing: border-box;
   width: 100%;
+  //border-bottom: 2px outset #eef;
```
Comment (emmaadesile) [inline src/components/Header/header.module.scss]: Please remove commented out code. line 10

**Review by emmaadesile — CHANGES_REQUESTED** [review]

**Review by emmaadesile — APPROVED** [review]

**Review by emekafredy — APPROVED** [review]
LGTM

**Review by johnkingzy — APPROVED** [review]
Nice one!

**Pull request merged**